plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -266,8 +275,41 @@ private Client client() {
     @FunctionalInterface
     interface ProcessorFactoryProvider {
 
-        Processor.Factory get(Environment environment);
+        Processor.Factory get(Environment environment, TemplateService templateService);
+
+    }
+
+    static class InternalTemplateService implements TemplateService {
+
+        public static final ScriptContext.Plugin INSTANCE = new ScriptContext.Plugin("elasticsearch-ingest", "ingest");
+
+        private final ScriptService scriptService;
+
+        InternalTemplateService(ScriptService scriptService) {
+            this.scriptService = scriptService;
+        }
 
+        @Override
+        public Template innerCompile(String template) {
+            Script script = new Script(template, ScriptService.ScriptType.INLINE, "mustache", Collections.emptyMap());
```

Comment:
no :) because `MustacheScriptEngineService` class is part of the lang-mustache module and we can't depend on it. Maybe we should add a constant in the `InternalTemplateService`?